core/lib/dal/src/eth_sender_dal.rs

```diff
@@ -686,26 +636,53 @@ impl EthSenderDal<'_, '_> {
         .context("count field is missing")
     }
 
-    pub async fn clear_failed_transactions(&mut self) -> sqlx::Result<()> {
+    pub async fn clear_failed_transactions(&mut self) -> anyhow::Result<()> {
+        // we don't allow in-flight txs as they might be confirmed on Ethereum, but not in db
+        if self.count_all_inflight_txs().await.unwrap() != 0 {
+            return Err(anyhow!(
+                "There are still some in-flight txs, cannot proceed. \
+            Please wait for eth-sender to process all in-flight txs and try again!"
+            ));
+        }
```

Comment:
This logic seems to be too high-level for DAL method. I would rather check `count_all_inflight_txs  == 0` on block reverter side. Also, it makes sense to either change method name (as it no longer only clears failed transactions but also removes unsent ones) or move removing unsent ones to a different method. And maybe `#[doc(hidden)]` + comment "this method should be used only by block-reverter" wouldn't hurt

Reply:
I've moved the check to block reverter, I've also changed the method name, it's super verbose now, but I still think it's a change for better